src/libData/AccountData/Transaction.h

```diff
@@ -134,11 +136,23 @@ class Transaction : public SerializableDataBlock {
   /// Returns the sender's Address
   Address GetSenderAddr() const;
 
-  /// Returns the transaction amount.
-  const uint128_t& GetAmount() const;
+  /// Returns the transaction amount in Qa.
+  const uint128_t GetAmountQa() const;
+
+  /// Returns the transaction amount in Wei.
+  const uint128_t GetAmountWei() const;
+
+  /// Returns the transaction amount in raw units regardless of Qa or Wei.
+  const uint128_t& GetAmountRaw() const;
```

Comment:
Here you return with &, but the other Getters don't They are all multi-precision objects and should be returned by ref.

Reply:
For Wei it returns temporary objects, so can't be by ref.   128bit is cheap in any case.